### PR TITLE
chore: Add spec for flux doc comments

### PR DESCRIPTION
**Pull request by scbrickley**
Adds a document containing the specification for the new Flux doc comments, as per #3770

There were a lot of comments on the PR with the rough draft of this document. I opened this new PR with the updated version to make any future conversations in the comments easier to follow.

[Original PR here.](https://github.com/influxdata/flux/pull/3786)

### CODE REVIEW

**Comment by BarWeiner1** [comment]
When we are writing the parameter list for various functions, should we specify their type? The docs website has this functionality so it might be worthwhile to include.

**Comment by nathanielc** [comment]
@BarWeiner1 
> When we are writing the parameter list for various functions, should we specify their type? The docs website has this functionality so it might be worthwhile to include.

No, this is something we can get automatically from type inference. We don't want the author of the docs to have to duplicate that information.

File: libflux/flux/FLUXDOC.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+To allow for this, Flux doc comments are formatted so that the doc comment parser
+can easily distinguish the short-form content from the long-form content. In most
+cases throughout this document, these will be referred to as the "headline" and
+the "description", respectively.
+
+## Writing Flux Doc Comments
+
+Flux will treat any uninterrupted series of line comments immediately preceding
+a function definition, package clause, or option declaration, as a doc comment.
```
Comment (nathanielc) [inline libflux/flux/FLUXDOC.md:33]: This should read:

> Flux will treat any uninterrupted series of line comments immediately preceding
an assignment statement, package clause, or option declaration, as a doc comment.

Packages do not exclusively define functions but can define values of any type.

**Review by nathanielc — COMMENTED** [review]
Looks great, I made some notes about allowing for non function values in packages. I'll add a commit to this branch to clarify what I am thinking.

File: libflux/flux/FLUXDOC.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+headline should read as one complete sentence.
+
+Follow the headline with a blank line comment.
+
+_Example:_
+
+Here's what the doc comment might look like for the pi constant in the math package.
+
+```
+// pi is a floating point representation of the mathematical constant pi.
```
Comment (pierwill) [inline libflux/flux/FLUXDOC.md:89]: Will we allow links? 

```
// pi is a floating point representation of the mathematical constant [pi](https://en.wikipedia.org/wiki/Pi).

**Review by pierwill — COMMENTED** [review]
Love this great work! I've made some comments in this PR, and have a submitted https://github.com/influxdata/flux/pull/3816. No need to merge that PR if it's too heavy-handed. Just thought it'd be easier to offer feedback that way!

File: libflux/flux/FLUXDOC.md (hunk 2 above)
Comment (nathanielc) [reply libflux/flux/FLUXDOC.md:89]: Yes

**Review by nathanielc — APPROVED** [review]

**Pull request merged**